physicsnemo/models/domino/model.py

```diff
@@ -55,55 +51,80 @@ def calculate_pos_encoding(nx, d=8):
+    """
+    return sdf / (0.4 + torch.abs(sdf))
 
 
 class BQWarp(nn.Module):
-    """Warp based ball-query layer"""
+    """
+    Warp-based ball-query layer for finding neighboring points within a specified radius.
+
+    This layer uses an accelerated ball query implementation to efficiently find points
+    within a specified radius of query points.
+    """
 
     def __init__(
         self,
-        input_features,
-        grid_resolution=[256, 96, 64],
-        radius=0.25,
-        neighbors_in_radius=10,
+        input_features: int,
+        grid_resolution=None,
+        radius: float = 0.25,
+        neighbors_in_radius: int = 10,
     ):
+        """
+        Initialize the BQWarp layer.
+
+        Args:
+            input_features: Number of feature dimensions for input points
+            grid_resolution: Resolution of the grid in each dimension [nx, ny, nz]
+            radius: Radius for ball query operation
+            neighbors_in_radius: Maximum number of neighbors to return within radius
+        """
         super().__init__()
+        if grid_resolution is None:
+            grid_resolution = [256, 96, 64]
         self.ball_query_layer = BallQueryLayer(neighbors_in_radius, radius)
         self.grid_resolution = grid_resolution
 
-    def forward(self, x, p_grid, reverse_mapping=True):
+    def forward(
+        self, x: torch.Tensor, p_grid: torch.Tensor, reverse_mapping: bool = True
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        """
+        Performs ball query operation to find neighboring points and their features.
+
+        This method uses the Warp-accelerated ball query implementation to find points
+        within a specified radius. It can operate in two modes:
+        - Forward mapping: Find points from x that are near p_grid points (reverse_mapping=False)
+        - Reverse mapping: Find points from p_grid that are near x points (reverse_mapping=True)
+
+        Args:
+            x: Tensor of shape (batch_size, num_points, 3+features) containing point coordinates
+               and their features
+            p_grid: Tensor of shape (batch_size, grid_x, grid_y, grid_z, 3) containing grid point
+                   coordinates
+            reverse_mapping: Boolean flag to control the direction of the mapping:
+                            - True: Find p_grid points near x points
+                            - False: Find x points near p_grid points
+
+        Returns:
+            tuple containing:
+                - mapping: Tensor containing indices of neighboring points
+                - outputs: Tensor containing coordinates of the neighboring points
+        """
         batch_size = x.shape[0]
-        nx, ny, nz = (
-            self.grid_resolution[0],
-            self.grid_resolution[1],
-            self.grid_resolution[2],
-        )
+        nx, ny, nz = self.grid_resolution
 
         p_grid = torch.reshape(p_grid, (batch_size, nx * ny * nz, 3))
```

Comment:
Do we need to supply nx/ny/nz (and, grid_resolution in the constructor)?  This should work on arbitrary point clouds, the only requirements should be that the shape is [batch_size, (anything), 3] and the forward function could quietly reshape appropriately.